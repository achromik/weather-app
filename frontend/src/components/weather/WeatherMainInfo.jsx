import React from 'react';
import { connect } from 'react-redux';
import styled from 'styled-components';

import { colors } from 'src/config/variables';
import { Wrapper } from 'src/common/components/Wrapper';
import { weatherDataSelector } from 'src/features/weather/selectors/weatherSelectors';
import { TemperatureWithIcon } from 'src/components/weather/TemperatureWhitIcon';

export const Header = styled.p`
    font-size: 1.5rem;
    font-weight: 600;
    color: ${colors.primary};
`;

export const WeatherMainInfoRaw = ({ cityName, country, weather, temp }) => (
    <Wrapper>
        <Header>
            {cityName}, {country}
        </Header>
        <TemperatureWithIcon temp={temp} icon={weather.icon} />
    </Wrapper>
);

const mapStateToProps = state => ({
    temp: weatherDataSelector(state).main,
    cityName: weatherDataSelector(state).name,
    country: weatherDataSelector(state).sys.country,
    weather: weatherDataSelector(state).weather[0],
});

export const WeatherMainInfo = connect(mapStateToProps)(WeatherMainInfoRaw);
