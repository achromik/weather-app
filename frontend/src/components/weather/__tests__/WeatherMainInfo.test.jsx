import React, { Children } from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import { WeatherMainInfoRaw } from '../WeatherMainInfo';

const mockProps = {
    weather: {
        id: 800,
        main: 'Clear',
        description: 'clear sky',
        icon: '01d',
    },
    temp: {
        temp: 25.02,
        pressure: 1006,
        humidity: 73,
        temp_min: 22.78,
        temp_max: 27.22,
    },
    country: 'PL',
    cityName: 'Warsaw',
};

describe('WeatherMainInfo', () => {
    it('renders', () => {
        const wrapper = shallow(<WeatherMainInfoRaw {...mockProps} />);
        expect(shallowToJson(wrapper)).toMatchSnapshot();
    });

    it('renders with city name and country', () => {
        const wrapper = shallow(<WeatherMainInfoRaw {...mockProps} />);
        expect(
            wrapper
                .children()
                .at(0)
                .text(),
        ).toBe('Warsaw, PL');
    });
});
