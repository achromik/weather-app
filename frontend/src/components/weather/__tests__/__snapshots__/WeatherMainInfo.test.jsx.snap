// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeatherMainInfo renders 1`] = `
<styled.div>
  <styled.p>
    Warsaw
    , 
    PL
  </styled.p>
  <TemperatureWithIcon
    icon="01d"
    temp={
      Object {
        "humidity": 73,
        "pressure": 1006,
        "temp": 25.02,
        "temp_max": 27.22,
        "temp_min": 22.78,
      }
    }
  />
  <styled.p>
    Clear sky
  </styled.p>
</styled.div>
`;
